pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartitionUpsertMetadataManager.java

```diff
@@ -193,12 +175,6 @@ public GenericRow updateRecord(IndexSegment segment, RecordInfo recordInfo, Gene
             if (recordInfo._comparisonValue.compareTo(currentRecordLocation.getComparisonValue()) >= 0) {
               IndexSegment currentSegment = currentRecordLocation.getSegment();
               int currentDocId = currentRecordLocation.getDocId();
-              if (_partialUpsertHandler != null) {
```

Comment:
I think we shall keep the partial upsert merge logic, as the groovy-based partial upsert logic can trigger some default logic even for newly ingested record

Reply:
The partial upsert merge logic is separated as a separate method `updateRecord()`. This PR will keep the current functionality of partial upsert, but fix the inconsistency issue mentioned above

Reply:
I see. So you extracted the update logic and perform it before adding the record.